src/algebra/big_operators/fin.lean

```diff
@@ -87,6 +84,10 @@ theorem prod_univ_cast_succ [comm_monoid β] {n : ℕ} (f : fin (n + 1) → β)
   ∏ i, f i = (∏ i : fin n, f i.cast_succ) * f (last n) :=
 by simpa [mul_comm] using prod_univ_succ_above f (last n)
 
+@[to_additive] lemma prod_cons [comm_monoid β] {n : ℕ} (x : β) (f : fin n → β) :
+  ∏ i : fin n.succ, (cons x f : fin n.succ → β) i = x * ∏ i : fin n, f i :=
```

Comment:
I wonder if this is more useful delta-reduced:
```suggestion
  finset.univ.prod (cons x f : fin n.succ → β) = x * ∏ i : fin n, f i :=
```
Can `rw` use either version to rewrite the other?

Reply:
Either version can indeed rewrite the other one.